mongo/mongo.go

```diff
@@ -83,3 +83,40 @@ func (m *Mongo) GetRole(ctx context.Context, id string) (*models.Role, error) {
 
 	return &role, nil
 }
+
+// GetRoles retrieves all role documents from Mongo, according to the provided limit and offset.
+//Offset and limit need to  be positive or zero. Zero limit is equivalent to no limit and all items starting at the offset will be returned.
+func (m *Mongo) GetRoles(ctx context.Context, offset, limit int) (*models.Roles, error) {
+	if offset < 0 || limit < 0 {
+		return nil, apierrors.ErrLimitAndOffset
+	}
+
+	s := m.Session.Copy()
+	defer s.Close()
+	log.Event(ctx, "getting roles", log.INFO)
+
+	roles := s.DB(m.Database).C(m.Collection).Find(nil)
+	totalCount, err := roles.Count()
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, apierrors.ErrRoleNotFound
+		}
+		return nil, err
+	}
+
+	results := []models.Role{}
+	iter := roles.Skip(offset).Limit(limit).Iter()
```

Comment:
Question: What happens if `offset > totalCount` ? I've had a quick look at the `Skip` documentation but it doesn't state what happens in this case. 

https://pkg.go.dev/github.com/globalsign/mgo#Query.Skip

Is this worth writing a test case for?

Reply:
This is a really good point, but I am not sure how I would write a test case for this scenario as currently all the responses from this method are mocked. I will have a look to see what I can find.

Reply:
You could easily test this locally and logging the error that gets returned, my assumption is that it would give you 0 results - possibly not found error

Reply:
@daiLlew we tested this locally by making a request with an offset higher than the number of documents stored in mongo. The iterator doesn't through an error but returns an empty array

Reply:
👍. Sorry for being pedantic but I think given that no one could say for certain what would happen it was worth taking the time to verify the behaviour. Thanks.